docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_in_vpc.mdx

```diff
@@ -23,6 +23,206 @@ CBP
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a VPC**:
+   If you don't already have a VPC, you need to create one. This step ensures that you have a VPC to launch your Redshift cluster into.
+   ```python
+   import boto3
+
+   ec2 = boto3.client('ec2')
+
+   # Create a VPC
+   response = ec2.create_vpc(
+       CidrBlock='10.0.0.0/16'
+   )
+   vpc_id = response['Vpc']['VpcId']
+   print(f'Created VPC with ID: {vpc_id}')
+   ```
+
+3. **Create Subnets and Security Groups**:
+   Create subnets and security groups within the VPC to ensure that your Redshift cluster has the necessary network infrastructure.
+   ```python
+   # Create a subnet
+   subnet_response = ec2.create_subnet(
+       VpcId=vpc_id,
+       CidrBlock='10.0.1.0/24'
+   )
+   subnet_id = subnet_response['Subnet']['SubnetId']
+   print(f'Created Subnet with ID: {subnet_id}')
+
+   # Create a security group
+   sg_response = ec2.create_security_group(
+       GroupName='RedshiftSG',
+       Description='Security group for Redshift',
+       VpcId=vpc_id
+   )
+   sg_id = sg_response['GroupId']
+   print(f'Created Security Group with ID: {sg_id}')
+   ```
+
+4. **Launch Redshift Cluster within the VPC**:
+   Use the VPC ID, subnet ID, and security group ID to launch the Redshift cluster within the VPC.
+   ```python
+   redshift = boto3.client('redshift')
+
+   # Launch Redshift cluster within the VPC
+   cluster_response = redshift.create_cluster(
+       ClusterIdentifier='my-redshift-cluster',
+       NodeType='dc2.large',
+       MasterUsername='masteruser',
+       MasterUserPassword='MasterUserPassword1!',
+       ClusterType='single-node',
+       VpcSecurityGroupIds=[sg_id],
+       ClusterSubnetGroupName='my-subnet-group'
+   )
+   print(f'Launched Redshift Cluster with ID: {cluster_response["Cluster"]["ClusterIdentifier"]}')
+   ```
+
+By following these steps, you ensure that your Redshift clusters are launched within a VPC, thereby preventing the misconfiguration of launching them outside a VPC.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "VPC ID" column. If a cluster has a VPC ID listed, it is launched within a VPC. If the "VPC ID" column is blank or not listed, the cluster is not launched within a VPC, indicating a misconfiguration.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>